### PR TITLE
Yuqiang/prepaid removedusers

### PR DESCRIPTION
**Pull request by smallst**
the key updates are schemas.
but add a simple modal to show the changes.

need help in naming the `active students` and `inactive students` (maybe also add description here so that everyone can understand it clearly)

### CODE REVIEW

File: app/views/teachers/LicenseStatsModal.coffee [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+      r.userID
+    )
+    @propsData =
+      hide: () => @hide()
+      loading: { finished: false }
+      redeemers: @redeemers
+      removedRedeemers: @removedRedeemers
+
+    @supermodel.resetProgress()
+    userNameRequest = @supermodel.addRequestResource 'user_names', {
```
Comment (mrfinch) [inline app/views/teachers/LicenseStatsModal.coffee:28]: created hook in modal seems better way to do this. Any reason we are doing it here?

File: app/views/teachers/components/LicenseStatsModal.vue [hunk 2]
```diff
@@ -0,0 +1,84 @@
+<script>
+const moment = require('moment')
+export default Vue.extend({
+  name: 'LicenseStatsModal',
+  props: ['hide', 'loading', 'redeemers', 'removedRedeemers'],
+  computed: {
+    moment () {
```
Comment (mrfinch) [inline app/views/teachers/components/LicenseStatsModal.vue:7]: could be inside methods too but just nitpick

File: app/views/teachers/components/LicenseStatsModal.vue [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  .user {
+    &:nth-child(2n+1) {
+      background-color: #ebebeb;
+    }
+    &:nth-child(2n) {
+      background-color: #f5f5f5;
+    }
+  }
+}
+</style>
```
Comment (mrfinch) [inline app/views/teachers/components/LicenseStatsModal.vue:84]: newline

File: app/views/teachers/components/LicenseStatsModal.vue [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    moment () {
+      return moment
+    }
+  }
+})
+
+</script>
+
+<template lang="pug">
+#modal-base-flat
```
Comment (mrfinch) [inline app/views/teachers/components/LicenseStatsModal.vue:16]: can't we use Modal component from `app/components/common/Modal` instead?

**Comment by mrfinch** [comment]
There should be some message shown when redeemers and deletedRedeemers are zero or don't show license stats button in that case

Screenshot 2022-09-07 at 1 25 41 PM

**Comment by smallst** [comment]
@mrfinch oh man, good catch. i even remember to add `empty results` for that school-admin-view but forget here.

File: app/views/teachers/components/LicenseStatsModal.vue (hunk 4 above)
Comment (smallst) [reply app/views/teachers/components/LicenseStatsModal.vue:16]: ok we have a new modal in `app/components/common` :joy: i believe i used it once but i totally forget it

Comment (smallst) [reply app/views/teachers/components/LicenseStatsModal.vue:16]: oh but i don't make `modal` display normally. seems it is transparency at all. so i still using the old modal 😂

**Review by mrfinch — APPROVED** [review]

**Pull request merged**